validator.js

```diff
@@ -0,0 +1,137 @@
+// Defining a function to display error message
+function printError(elemId, hintMsg) {
+    document.getElementById(elemId).innerHTML = hintMsg;
+}
+
+// Defining a function to validate form 
+function validateForm() {
```

Comment:
The code is a bit messy here due to duplications. 

Here you can see a common pattern that will help you find out and eliminate the duplication. 

For example: 
```jsx
var regex = /^[a-zA-Z\s]+$/;                
        if(regex.test(name) == false || name == "") {
            printError("nameErr", "Please enter a valid name(only alphabeths)");
        } else {
            printError("nameErr", "");
            nameErr = false;
        } 
```

The only difference between fields is the regex to be used for validation and the error message. 
You can use polymorphism here to remove the duplication code. 

For example:
```jsx
let validators = [
  new Validator('name', /^[a-zA-Z\s]+$/, 'Name is not valid')
]
```

So when code iterates over validator objects it will validate all registered fields.

Reply:
Okay, I'll remake that.

Reply:
I have removed some of the duplicate code as I try to shorten it with polymorphism.

Reply:
I removed dublicate code.